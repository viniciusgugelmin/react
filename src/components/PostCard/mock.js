export const PostCardPropsMock = {
  title: 'Lorem ipsum title',
  body: 'Lorem ipsum body',
  id: 1,
  cover: 'img/lorem-ipsum-img.png',
};
