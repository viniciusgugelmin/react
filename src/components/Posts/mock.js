export const PostsMock = [
  {
    id: 1,
    title: 'Lorem ipsum title 1',
    body: 'Lorem ipsum body 1',
    cover: 'img/img1.png',
  },
  {
    id: 2,
    title: 'Lorem ipsum title 2',
    body: 'Lorem ipsum body 2',
    cover: 'img/img2.png',
  },
  {
    id: 3,
    title: 'Lorem ipsum title 3',
    body: 'Lorem ipsum body 3',
    cover: 'img/img3.png',
  },
];
