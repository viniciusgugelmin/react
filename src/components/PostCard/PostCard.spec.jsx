import { render, screen } from '@testing-library/react';
import { PostCard } from './index';
import { PostCardPropsMock as props } from './mock';

describe('<PostCard />', function () {
  it('should render correctly', () => {
    render(<PostCard {...props} />);

    expect(screen.getByRole('img', { name: props.title })).toHaveAttribute('src', props.cover);
    expect(screen.getByRole('img', { name: props.title })).toHaveAttribute('alt', props.title);
    expect(screen.getByRole('heading', { name: props.title })).toBeInTheDocument();
    expect(screen.getByText(props.body)).toBeInTheDocument();
  });

  it('should match snapchot', () => {
    const { container } = render(<PostCard {...props} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
