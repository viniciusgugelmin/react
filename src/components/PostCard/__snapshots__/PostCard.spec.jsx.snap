// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapchot 1`] = `
<div
  class="post"
>
  <img
    alt="Lorem ipsum title"
    src="img/lorem-ipsum-img.png"
  />
  <div
    class="post-content"
  >
    <h2>
      Lorem ipsum title
    </h2>
    <p>
      Lorem ipsum body
    </p>
  </div>
</div>
`;
